metals/src/main/scala/scala/meta/internal/metals/codeactions/PatternMatchRefactor.scala

```diff
@@ -0,0 +1,82 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.Term
+import scala.meta.internal.metals.CodeAction
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+
+import org.eclipse.{lsp4j => l}
+
+class PatternMatchRefactor(trees: Trees) extends CodeAction {
+
+  override val kind: String = l.CodeActionKind.RefactorRewrite
+
+  private def convert(
+      patternMatch: Term.Match,
+      path: AbsolutePath
+  ): Seq[l.CodeAction] = {
+    patternMatch.expr match {
+      case _: Term.Placeholder =>
+        val range = patternMatch.pos.toLSP
+        val indentSize = range.getStart.getCharacter
+        // in order to replace current indent in text edit
+        range.getStart.setCharacter(0)
+
+        def stripMultilineCaseIndent(caze: meta.Case): String = {
+          val lines = caze.toString.split("\n")
+          (lines.take(1) ++ lines.drop(1).map(_.stripPrefix("  ")))
+            .mkString("\n")
+        }
+
+        val indent = " " * indentSize
+        val text = patternMatch.cases
+          .map(c => s"$indent${stripMultilineCaseIndent(c)}")
+          .mkString("\n")
+
+        val edits = List(new l.TextEdit(range, text))
+        val codeAction = new l.CodeAction()
+        codeAction.setTitle(PatternMatchRefactor.convertPatternMatch)
+        codeAction.setKind(this.kind)
+        codeAction.setEdit(
+          new l.WorkspaceEdit(
+            Map(path.toURI.toString -> edits.asJava).asJava
+          )
+        )
+        Seq(codeAction)
+      case _ => Seq.empty
+    }
+  }
+
+  override def contribute(params: l.CodeActionParams, token: CancelToken)(
+      implicit ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = {
+
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+    val x = trees
+      .findLastEnclosingAt[Term.Block](path, range.getStart())
```

Comment:
Will this work if we have multiple blocks? We might want to add an additional condition about the tree.